lib/dry/types/struct.rb

```diff
@@ -87,10 +89,9 @@ def [](name)
       end
 
       def to_hash
-        self.class.schema.keys.each_with_object({}) { |key, result|
-          value = self[key]
-          result[key] = value.respond_to?(:to_hash) ? value.to_hash : value
-        }
+        self.class.schema.keys.each_with_object({}) do |key, result|
+          result[key] = Hashify[self[key]]
```

Comment:
D'oh! You're right about instantiations. I should have implemented hashify as a utility module (and I will).

I couldn't drop external hashifier and implement the functionality in the method because we need going through many nested arrays (or sets etc.).

Their items do not necessarily implement `#to_hash` by themselves, so we need to check `to_hash` and `map` at any layer by some external visitor/hashifier.